pipelines/validate_vset.py

```
@@ -0,0 +1,894 @@
+    missing = [key for key in KIND_PAYLOAD_KEYS[kind] if key not in payload]
+    if missing:
+        return [
+            VSetValidationError(
+                "vset.payload_invalid",
+                f"{kind} payload missing {missing}",
+            )
+        ]
+    return []
+
+
+def validate_record(
```

Ruff (0.16.2)

[warning] Too many branches (39 > 12)

(PLR0912)